atat_internal_api.yaml

```
@@ -0,0 +1,2197 @@
+          "code": "OTHER"
+          "message": "A valid request body must be provided"
+        }
+  DEFAULT_4XX:
+    # Use the original status code for 4XX errors
+    responseParameters:
+      gatewayresponse.header.Access-Control-Allow-Headers: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Methods: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Origin: '''*'''
+    responseTemplates:
+      application/json: |
+        {
+          "code": "OTHER",
+          "message": "The request is not valid"
+        }
+  DEFAULT_5XX:
+    # Rewrite all 5XX errors as a 500
+    statusCode: 500
+    responseParameters:
+      gatewayresponse.header.Access-Control-Allow-Headers: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Methods: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Origin: '''*'''
+    responseTemplates:
+      application/json: |
+        {
+          "code": "OTHER",
+          "message": "There was an error processing the request"
+        }
+  # EXPIRED_TOKEN is handled by DEFAULT_4XX
+  # INTEGRATION_FAILURE is handled by DEFAULT_5XX
+  # INTEGRATION_TIMEOUT is handled by DEFAULT_5XX
+  # INVALID_API_KEY is handled by DEFAULT_4XX
+  # INVALID_SIGNATURE is handled by DEFAULT_4XX
+  MISSING_AUTHENTICATION_TOKEN:
+    # Treat this the same way as a 404/403 would be treated
+    statusCode: 403
+    responseParameters:
+      gatewayresponse.header.Access-Control-Allow-Headers: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Methods: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Origin: '''*'''
+    responseTemplates:
+      application/json: |
+        {
+          "code": "OTHER",
+          "message": "The requested resource could not be accessed"
+        }
+  # QUOTA_EXCEEDED is handled by DEFAULT_4XX
+  # REQUEST_TOO_LARGE is handled by DEFAULT_4XX
+  RESOURCE_NOT_FOUND:
+    # Mask gateway-level 404s as a 403 to prevent "guessing" whether
+    # a particular endpoint exists or not.
+    statusCode: 403
+    responseParameters:
+      gatewayresponse.header.Access-Control-Allow-Headers: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Methods: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Origin: '''*'''
+    responseTemplates:
+      application/json: |
+        {
+          "code": "OTHER",
+          "message": "The requested resource could not be accessed"
+        }
+  # THROTTLED is handled by DEFAULT_4XX
+  UNAUTHORIZED:
+    statusCode: 401
+    responseParameters:
+      gatewayresponse.header.Access-Control-Allow-Headers: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Methods: '''*'''
+      gatewayresponse.header.Access-Control-Allow-Origin: '''*'''
+    responseTemplates:
+      application/json: |
+        {
+          "code": "OTHER",
+          "message": "Authentication is required for this endpoint"
+        }
+  # UNSUPPORTED_MEDIA_TYPE is handled by DEFAULT_4XX
+  # WAF_FILTERED is handled by DEFAULT_4XX
```

```suggestion
  # WAF_FILTERED is handled by DEFAULT_4XX

```

An EOL at the end of the file keeps things a little cleaner when interacting with CLI and other tools.